de.dlr.sc.virsat.model.extension.cefx.ui/src/de/dlr/sc/virsat/model/extension/cefx/ui/importWizards/CometDataFetcher.java

```diff
@@ -0,0 +1,176 @@
+
+
+package de.dlr.sc.virsat.model.extension.cefx.ui.importWizards;
+
+import cdp4common.engineeringmodeldata.ElementDefinition;
+import cdp4common.engineeringmodeldata.ElementUsage;
+import cdp4common.engineeringmodeldata.EngineeringModel;
+import cdp4common.engineeringmodeldata.Iteration;
+import cdp4common.engineeringmodeldata.Parameter;
+import cdp4common.engineeringmodeldata.ParameterSwitchKind;
+import cdp4common.engineeringmodeldata.ParameterValueSet;
+import cdp4common.sitedirectorydata.DomainOfExpertise;
+import cdp4dal.Session;
+import cdp4dal.SessionImpl;
+import cdp4dal.dal.Credentials;
+import cdp4servicesdal.CdpServicesDal;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.MessageBox;
+import org.eclipse.swt.widgets.Tree;
+import org.eclipse.swt.widgets.TreeItem;
+
+import java.net.URI;
+import java.util.Optional;
+
+/**
+ * The CometDataFetcher class is responsible for fetching and populating 
+ * data from a COMET system into a SWT  Tree widget.
+ */
+public class CometDataFetcher {
+
+    private final Credentials credentials;
+    private final Session session;
+    private final URI uri;
+
+    /**
+     * Constructs a new CometDataFetcher instance.
+     * 
+     * @param url      The URL of the COMET system.
+     * @param username The username for authentication.
+     * @param password The password for authentication.
+     */
+    public CometDataFetcher(String url, String username, String password) {
+        this.uri = URI.create(url);
+        CdpServicesDal dal = new CdpServicesDal();
+        this.credentials = new Credentials(username, password, this.uri, null);
+        this.session = new SessionImpl(dal, this.credentials);
+    }
+
+    /**
+     * Initiates the process to fetch data and populate the provided Tree.
+     */
+    public void fetchData(Tree tree) {
+        new Thread(() -> {
+            try {
+                openSession();
+                Display.getDefault().asyncExec(() -> populateTreeWithIteration(tree));
+            } catch (Exception e) {
+                handleException(e, tree);
+            }
+        }).start();
+    }
+
+    /**
+     * Opens the session for communication with the COMET system.
+     */
+    private void openSession() {
+        session.open().join();
+    }
+
+    /**
+     * Populates the provided Tree with data from the current iteration 
+     * of the engineering model.
+     * 
+     * @param tree The SWT Tree widget to populate.
+     */
+    private void populateTreeWithIteration(Tree tree) {
+        var siteDirectory = session.getAssembler().retrieveSiteDirectory();
+        var engineeringModelIid = siteDirectory.getModel().get(0).getEngineeringModelIid();
+        var iterationIid = siteDirectory.getModel().get(0).getIterationSetup().get(0).getIterationIid();
+        var domainOfExpertiseIid = siteDirectory.getModel().get(0).getActiveDomain().get(0).getIid();
```

Comment:
In the existing code, we use the first DomainOfExpertise as a default selection to retrieve data like the example of comet. This is a simplification assuming that the first domain is sufficient for the intended operation. If we need to access data from a different domain, the code would need to be extended to allow selection or iteration over the available domains rather than defaulting to the first one.